### PR TITLE
Update Powershell auto-generator for Swagger library

### PR DESCRIPTION
**Pull request by hyonholee**

### CODE REVIEW

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/VirtualMachineVMImage/VirtualMachineVMImageUpdateMethod.cs [hunk 1]
```diff
@@ -42,22 +42,22 @@ protected object CreateVirtualMachineVMImageUpdateDynamicParameters()
...
             pParameters.Attributes.Add(new ParameterAttribute
             {
                 ParameterSetName = "InvokeByDynamicParameters",
                 Position = 2,
-                Mandatory = true
+                Mandatory = false
             });
             pParameters.Attributes.Add(new AllowNullAttribute());
-            dynamicParameters.Add("VirtualMachineVMImageUpdateParameters", pParameters);
+            dynamicParameters.Add("VirtualMachineVMImageUpdateParameter", pParameters);
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/VirtualMachineVMImage/VirtualMachineVMImageUpdateMethod.cs:60]: It seems this is impacting the Hyak side. The code generator needs to know about whether it's processing Hyak/AutoRest, and we would still need Hyak for ASM.

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Automation/StringProcessingHelper.ps1 [hunk 2]
```diff
@@ -292,3 +292,25 @@ function Get-SplitTextLines
...
+        [Parameter(Mandatory = $true)]
+        [string]$noun
+    )
+
+    if ($noun -eq $null)
+    {
+        return $noun;
+    }
+
+    if ($noun.EndsWith("s"))
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Automation/StringProcessingHelper.ps1:313]: We may need a word processing function here, :-)

I think it's OK for now.

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs [hunk 3]
```diff
@@ -54,22 +54,22 @@ protected object CreateDNSServerAddDNSServerDynamicParameters()
...
             });
             pDeploymentName.Attributes.Add(new AllowNullAttribute());
             dynamicParameters.Add("DeploymentName", pDeploymentName);
 
-            var pAddress = new RuntimeDefinedParameter();
-            pAddress.Name = "Address";
-            pAddress.ParameterType = typeof(string);
-            pAddress.Attributes.Add(new ParameterAttribute
+            var pAddres = new RuntimeDefinedParameter();
+            pAddres.Name = "Addres";
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs]: It seems like a bug from the `Get-SingleNoun` function.

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs [hunk 4]
```diff
@@ -42,7 +42,7 @@ protected object CreateDNSServerAddDNSServerDynamicParameters()
             {
                 ParameterSetName = "InvokeByDynamicParameters",
                 Position = 1,
-                Mandatory = true
+                Mandatory = false
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs:45]: Why all RDFE cmdlets become non mandatory? Any side effect?

**Comment by huangpf** [comment]
Please check the RDFE tests: https://github.com/hyonholee/azure-powershell/blob/vmss/src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1#L169

Or try not to change the RDFE side.

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/VirtualMachineVMImage/VirtualMachineVMImageUpdateMethod.cs (hunk 1 above)
Comment (hyonholee) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/VirtualMachineVMImage/VirtualMachineVMImageUpdateMethod.cs:60]: I changed all plural parameter names to singular, to meet the powershell standard.  I updated the test code regarding this.

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs (hunk 4 above)
Comment (hyonholee) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/DNSServer/DNSServerAddDNSServerMethod.cs:45]: This is a side effect of update for CRP cmdlets.  But does it have a problem?

**Comment by huangpf** [comment]
LGTM.

**Pull request merged**